src/configuration/types.rs

```diff
@@ -261,6 +278,7 @@ pub struct Configuration {
   pub use_tabs: bool,
   pub new_line_kind: NewLineKind,
   pub quote_style: QuoteStyle,
+  pub quote_props: QuoteProps,
```

Comment:
Ooh yes, I'm a big fan of proper schemas for autocomplete.  I'll add that.

Reply:
@cspotcode no. It's a little complicated because someone can be using a dprint.json file with any kind of formatting plugin and the schema can change between plugin versions.

Right now, the dprint CLI asks the plugins what their schema URL is based on what plugins the user has specified in their dprint.json (ex. https://plugins.dprint.dev/dprint/dprint-plugin-typescript/0.62.2/schema.json) and then it provides that information to the vscode extension for example. The vscode extension then creates a new dynamically generated schema with that information in it.

So for example, here it figures out the "typescript", "json", and "markdown" keys based on what's specified in the "plugins" array:

![image](https://user-images.githubusercontent.com/1609021/153028814-88b17239-eea5-4c79-827a-b8f10a02114c.png)

Reply:
I see, that is a bit tricky.